### PR TITLE
Apply OpenRewrite assertj recipes

### PR DESCRIPTION
**Pull request by mfussenegger**
Applies the recipes from https://docs.openrewrite.org/recipes/java/testing/assertj
All executed via maven. For example:

    mvn -U org.openrewrite.maven:rewrite-maven-plugin:run \
      -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-testing-frameworks:RELEASE \
      -Drewrite.activeRecipes=org.openrewrite.java.testing.assertj.JUnitAssertEqualsToAssertThat

### CODE REVIEW

File: server/src/test/java/io/crate/planner/optimizer/OptimizerRuleSessionSettingProviderTest.java [hunk 1]
```diff
@@ -81,6 +81,6 @@ public void test_optimizer_rule_session_settings() {
 
         // Enable MergeFilters 'SET SESSION optimizer_merge_filters = true'
         sessionSetting.apply(sessionSettings, List.of(Literal.of(true)), eval);
-        assertThat(sessionSettings.excludedOptimizerRules().isEmpty(), is(true));
+        assertThat(sessionSettings.excludedOptimizerRules().isEmpty()).isEqualTo(true);
```
Comment (matriv) [inline server/src/test/java/io/crate/planner/optimizer/OptimizerRuleSessionSettingProviderTest.java:84]:
```suggestion
        assertThat(sessionSettings.excludedOptimizerRules().isEmpty()).isTrue();
```

File: server/src/test/java/io/crate/metadata/upgrade/IndexTemplateUpgraderTest.java [hunk 2]
```diff
@@ -116,11 +115,7 @@ public void testInvalidSettingIsRemovedForTemplateInCustomSchema() throws Except
         IndexTemplateUpgrader indexTemplateUpgrader = new IndexTemplateUpgrader();
         Map<String, IndexTemplateMetadata> result = indexTemplateUpgrader.apply(Collections.singletonMap(templateName, template));
 
-        assertThat(
-            "Outdated setting `index.recovery.initial_shards` must be removed",
-            result.get(templateName).settings().hasValue("index.recovery.initial_shards"),
-            is(false)
-        );
+        assertThat(result.get(templateName).settings().hasValue("index.recovery.initial_shards")).as("Outdated setting `index.recovery.initial_shards` must be removed").isEqualTo(false);
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/upgrade/IndexTemplateUpgraderTest.java:118]:
```suggestion
        assertThat(result.get(templateName).settings().hasValue("index.recovery.initial_shards")).as("Outdated setting `index.recovery.initial_shards` must be removed").isFalse();
```

File: server/src/test/java/io/crate/metadata/sys/SysSnapshotRestoreTableInfoTest.java [hunk 3]
```diff
@@ -60,14 +59,14 @@ public void test_convert_restore_in_progress_to_sys_restore_snapshot_info() {
             .snapshotsRestoreInProgress(new RestoreInProgress.Builder().add(entry).build())
             .iterator();
 
-        assertThat(restoreInProgressIt.hasNext(), is(true));
-        assertThat(restoreInProgressIt.next(), is(SysSnapshotRestoreInProgress.of(entry)));
-        assertThat(restoreInProgressIt.hasNext(), is(false));
+        assertThat(restoreInProgressIt.hasNext()).isEqualTo(true);
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/sys/SysSnapshotRestoreTableInfoTest.java:62]: same in this file.

File: server/src/test/java/io/crate/metadata/ColumnIdentTest.java [hunk 4]
```diff
@@ -69,24 +67,23 @@ public void testIsChildOf() throws Exception {
         ColumnIdent rootXY = new ColumnIdent("root", Arrays.asList("x", "y"));
         ColumnIdent rootYX = new ColumnIdent("root", Arrays.asList("y", "x"));
 
-        assertThat(root.isChildOf(root), is(false));
+        assertThat(root.isChildOf(root)).isEqualTo(false);
```
Comment (matriv) [inline server/src/test/java/io/crate/metadata/ColumnIdentTest.java:70]: same

File: server/src/test/java/io/crate/execution/engine/fetch/FetchCollectorTest.java [hunk 5]
```diff
@@ -41,12 +40,12 @@ public void test_sequential_docs_ids() {
         for (int i = start; i < start + 10; i++) {
             sequential.add(i);
         }
-        assertThat(FetchCollector.isSequential(sequential), is(true));
+        assertThat(FetchCollector.isSequential(sequential)).isEqualTo(true);
```
Comment (matriv) [inline server/src/test/java/io/crate/execution/engine/fetch/FetchCollectorTest.java:43]: same

File: server/src/test/java/io/crate/execution/engine/distribution/merge/BatchPagingIteratorTest.java [hunk 6]
```diff
@@ -146,7 +145,7 @@ public void testFinishPagingIteratorOnClose() {
         );
 
         iterator.close();
-        assertThat(pagingIterator.finishedCalled, Matchers.is(true));
+        assertThat(pagingIterator.finishedCalled).isEqualTo(true);
```
Comment (matriv) [inline server/src/test/java/io/crate/execution/engine/distribution/merge/BatchPagingIteratorTest.java:148]: same

File: server/src/test/java/io/crate/execution/dsl/projection/UpdateProjectionTest.java [hunk 7]
```diff
@@ -47,9 +47,9 @@ public void testEquals() throws Exception {
         UpdateProjection u2 = new UpdateProjection(
             Literal.of(1), new String[]{"foo"}, new Symbol[]{Literal.of(1)},new Symbol[]{new InputColumn(0, DataTypes.STRING)}, null, null);
 
-        assertThat(u2.equals(u1), is(true));
-        assertThat(u1.equals(u2), is(true));
-        assertThat(u1.hashCode(), is(u2.hashCode()));
+        assertThat(u2.equals(u1)).isEqualTo(true);
```
Comment (matriv) [inline server/src/test/java/io/crate/execution/dsl/projection/UpdateProjectionTest.java:50]: same

**Review by matriv — APPROVED** [review]
nice!
If you feel like satisfying my OCD :), you can replace the `isEqualTo(true/false)`.

**Comment by mfussenegger** [comment]
> If you feel like satisfying my OCD :), you can replace the isEqualTo(true/false).

Changed the, but you know you're not supposed to look that closely at automated refactors? :p

**Pull request merged**